compiler/rmi/rmi-interface/src/org/jetbrains/kotlin/rmi/DaemonParams.kt

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2010-2015 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.rmi
+
+import java.io.File
+import java.io.Serializable
+import java.lang.management.ManagementFactory
+import java.security.DigestInputStream
+import java.security.MessageDigest
+import kotlin.platform.platformStatic
+import kotlin.reflect.KMutableProperty1
+import kotlin.reflect.KProperty1
+
+
+public val COMPILER_JAR_NAME: String = "kotlin-compiler.jar"
+public val COMPILER_SERVICE_RMI_NAME: String = "KotlinJvmCompilerService"
+public val COMPILER_DAEMON_CLASS_FQN: String = "org.jetbrains.kotlin.rmi.service.CompileDaemon"
+public val COMPILE_DAEMON_DEFAULT_PORT: Int = 17031
+public val COMPILE_DAEMON_ENABLED_PROPERTY: String ="kotlin.daemon.enabled"
+public val COMPILE_DAEMON_JVM_OPTIONS_PROPERTY: String ="kotlin.daemon.jvm.options"
+public val COMPILE_DAEMON_OPTIONS_PROPERTY: String ="kotlin.daemon.options"
+public val COMPILE_DAEMON_CMDLINE_OPTIONS_PREFIX: String ="--daemon-"
+
+
+open class PropExtractor<C, V, P: KProperty1<C, V>>(val dest: C,
+                                                    val prop: P,
+                                                    val name: String,
+                                                    val convert: ((v: V) -> String?) = { it.toString() },
+                                                    val skipIf: ((v: V) -> Boolean) = { false },
+                                                    val mergeWithDelimiter: String? = null)
+{
+    constructor(dest: C, prop: P, convert: ((v: V) -> String?) = { it.toString() }, skipIf: ((v: V) -> Boolean) = { false }) : this(dest, prop, prop.name, convert, skipIf)
+    open fun extract(prefix: String = COMPILE_DAEMON_CMDLINE_OPTIONS_PREFIX): List<String> =
+            when {
+                skipIf(prop.get(dest)) -> listOf<String>()
+                mergeWithDelimiter != null -> listOf(prefix + name + mergeWithDelimiter + convert(prop.get(dest))).filterNotNull()
```

Comment:
we can get list with string which ends with "null" and `filterNotNull` is unnecessary here